tests/console/yast2_dns_server.pm

```diff
@@ -142,32 +105,62 @@ sub run {
         wait_screen_change { send_key 'alt-t' };
     }
     else {
-        change_service_configuration(after_reboot => {do_not_start => 'alt-a'});
+        change_service_configuration(
+            after_writing => {start        => 'alt-t'},
+            after_reboot  => {do_not_start => 'alt-a'}
+        );
+        send_key $cmd{apply_changes};
+        verify_service_configuration(status => 'active');
     }
-    send_key 'alt-o';
+    send_key $cmd{ok};
+
+    # Verify changes in console too
     assert_screen 'root-console', 180;
-    $self->assert_running(1);
-    $self->assert_enabled(0);
+    systemctl 'is-active named';
+    systemctl 'is-enabled named', expect_false => 1;
 
     #
-    # Fourth execution (tree-based interface)
+    # 4th run (tree-based interface)
     #
+    record_info '4th run',         '[tree-based interface] -> service inactive and disabled';
     script_run 'yast2 dns-server', 0;
     continue_info_network_manager_default;
-    if (is_sle('<15') || is_leap('<15.1')) {
+    if ($older_products) {
         assert_screen 'yast2-service-running-disabled', 90;
         # Stop the service
         send_key 'alt-s';
         assert_screen 'yast2-service-stopped-disabled';
     }
     else {
         change_service_configuration(after_writing => {stop => 'alt-t'});
+        send_key $cmd{apply_changes};
+        verify_service_configuration(status => 'inactive');    # verify changes in gui first
     }
-    # Finish
-    send_key 'alt-o';
+    send_key $cmd{ok};
+
+    # Verify changes in console too
     assert_screen 'root-console', 180;
-    $self->assert_running(0);
-    $self->assert_enabled(0);
+    systemctl 'is-active named',  expect_false => 1;
+    systemctl 'is-enabled named', expect_false => 1;
+
+    #
+    # 5th run (tree-based interface): only for new products as cancel do not revert changes in services status
+    #
+    unless ($older_products) {
```

Comment:
I suggest to use an early return, e.g. `return if $older_products;` and then just continue for newer products